configuration/template/stack.go

```diff
@@ -128,6 +135,88 @@ func MakeConfigAccessFuncs(confSvc ConfigurationService, varStack map[string]str
 	}
 }
 
+func MakeConfigAndRepoAccessFuncs(confSvc ConfigurationService, varStack map[string]string, workflowRepo repos.IRepo) map[string]interface{} {
+	return map[string]interface{} {
+		"GenerateDplSubworkflow": func (dplCommand string) string {
+			// jitDplGenerate takes a resolved dplCommand as an argument,
+			// generates the corresponding tasks and workflow
+			// and returns the resolved dplWorkflow as a string
+			jitDplGenerate := func(dplCommand string) (dplWorkflow string) {
+				const nMaxExpectedQcPayloads = 2
+				var metadata string
+
+				// Match any consul URL
+				re := regexp.MustCompile(`'consul-json://[^']*'`)
+				matches := re.FindAllStringSubmatch(dplCommand, nMaxExpectedQcPayloads)
+
+				// Concatenate the consul LastIndex for each payload in a single string
+				for _, match := range matches {
+					// Match any key under components
+					keyRe := regexp.MustCompile(`components/[^']*`)
+					consulKeyMatch := keyRe.FindAllStringSubmatch(match[0], 1)
+					consulKey := strings.SplitAfter(consulKeyMatch[0][0], "components/")
+
+					// And query for Consul for its LastIndex
+					newQ, err := componentcfg.NewQuery(consulKey[1])
+					_, lastIndex, err := confSvc.GetComponentConfigurationWithLastIndex(newQ)
+					if err != nil {
+						return fmt.Sprintf("JIT failed trying to query qc consul payload %s : " + err.Error(),
+							match)
+					}
+					metadata += strconv.FormatUint(lastIndex, 10)
+				}
+
+				var err error
+
+				// Generate a hash out of the concatenation of
+				// 1) The full DPL command
+				// 2) The LastIndex of each payload
+				hash := sha1.New()
+				hash.Write([]byte(dplCommand + metadata))
+				jitWorkflowName := "jit-" + hex.EncodeToString(hash.Sum(nil))
```

Comment:
Just a note for later: in order to avoid polluting the repo dir with JIT-generated files but still maintain filesystem semantics, we could output to a subdirectory in `/tmp` (which is usually a RAM mount anyway) and then use https://github.com/dschmidt/go-layerfs to keep using the workflow loader same as before. Just an idea...